bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ContentTypesPreferencePage.java

```diff
@@ -113,7 +112,7 @@ public ContentTypesPreferencePage() {
 
 	private Button addEditorAssociationButton;
 
-	private Set<Image> disposableEditorIcons = new HashSet<>();
+	private Collection<Image> disposableEditorIcons = new ArrayList<>();
```

Comment:
What about IdentityHashMap#keySet()? I think that's what actually is desired here...